flutter/lib/src/navigation/time_to_initial_display_tracker.dart

```diff
@@ -0,0 +1,119 @@
+// ignore_for_file: invalid_use_of_internal_member
+
+import 'dart:async';
+
+import 'package:meta/meta.dart';
+import '../integrations/integrations.dart';
+
+import '../../sentry_flutter.dart';
+import '../frame_callback_handler.dart';
+import '../sentry_flutter_measurement.dart';
+
+@internal
+class TimeToInitialDisplayTracker {
+  static final TimeToInitialDisplayTracker _instance =
+      TimeToInitialDisplayTracker._();
+
+  factory TimeToInitialDisplayTracker(
+      {FrameCallbackHandler? frameCallbackHandler}) {
+    if (frameCallbackHandler != null) {
+      _instance._frameCallbackHandler = frameCallbackHandler;
+    }
+    return _instance;
+  }
+
+  TimeToInitialDisplayTracker._();
+
+  FrameCallbackHandler _frameCallbackHandler = DefaultFrameCallbackHandler();
+  bool _isManual = false;
+  Completer<DateTime>? _trackingCompleter;
+  DateTime? _endTimestamp;
+
+  /// This endTimestamp is needed in the [TimeToFullDisplayTracker] class
+  @internal
+  DateTime? get endTimestamp => _endTimestamp;
+
+  Future<void> trackRegularRoute(ISentrySpan transaction,
+      DateTime startTimestamp, String routeName) async {
```

Comment:
you can add a nullable `endTimestamp` and a nullable `origin` parameters, null on regular routes, and have the `trackAppStart` just call this method, as they are nearly identical.
Or you can create a new private method called by both